core/domain/state_domain.py

```diff
@@ -173,6 +174,187 @@ def validate(self):
         self.hint_content.validate()
 
 
+class Image(object):
+    """Value object representing the image for the state."""
+    def __init__(self, src, placeholder, author_id, instructions):
+        """Constructs an Image domain object.
+
+        Args:
+            src: str. The source of an image.
+            placeholder: bool. Is image a placeholder.
+            author_id: str. Id of an author which uploads an image.
+            instructions: str. Instructions for the image which needs in place
+                of a placeholder.
+        """
+        self.src = src
+        self.placeholder = placeholder
+        self.author_id = author_id
+        self.instructions = instructions
+        self.validate()
+
+    def to_dict(self):
+        """Returns a dict representing this Image domain object.
+
+        Returns:
+            dict. A dict, mapping all fields of Image instance.
+        """
+        image = {}
+        image['src'] = self.src
+        image['placeholder'] = self.placeholder
+        image['author_id'] = self.author_id
+        image['instructions'] = self.instructions
+        return image
```

Comment:
Please follow the existing pattern of the to_dict method.

Reply:
done